ocaml/xapi/import.ml

```diff
@@ -1365,80 +1369,94 @@ end
 module VBD : HandlerTools = struct
   type precheck_t = Found_VBD of API.ref_VBD | Skip | Create of API.vBD_t
 
-  let precheck __context config rpc session_id state x =
+  let precheck __context config _rpc _session_id state x =
     let vbd_record = API.vBD_t_of_rpc x.snapshot in
     let get_vbd () =
-      Client.VBD.get_by_uuid ~rpc ~session_id ~uuid:vbd_record.API.vBD_uuid
+      (* Use an internal DB call - this avoids raising an exception and logging
+         the backtrace internally in case of a (reasonably expected) absence of
+         the object with this UUID *)
+      Db.VBD.get_by_uuid_opt ~__context ~uuid:vbd_record.API.vBD_uuid
     in
-    let vbd_exists () =
-      try
-        ignore (get_vbd ()) ;
-        true
-      with _ -> false
+    let vbd_opt =
+      (* If there's already a VBD with the same UUID and we're preserving UUIDs, use that one. *)
+      if config.full_restore then (
+        match get_vbd () with
+        | Some x ->
+            Some x
+        | None ->
+            info
+              "Did not find an already existing VBD with the same uuid=%s, try \
+               to create a new one"
```

Comment:
When would it try to create a new one? When actually importing the new VM?

Reply:
Yes, with a dummy VBD being created in the handler:
```
    match precheck_result with
   ...
    | Create _ ->
        let dummy_vbd = Ref.make () in
        state.table <- (x.cls, x.id, Ref.string_of dummy_vbd) :: state.table

```